plugins/out_s3/s3.c

```diff
@@ -1202,17 +1172,12 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
             else {
                 s3_retry_warn(ctx, (char *) chunk->fsf->meta_buf, m_upload->input_name,
                               chunk->create_time, FLB_TRUE);
-                s3_store_file_unlock(chunk);
```

Comment:
a few lines above here we call `s3_decrement_index(ctx);` after UploadPart failure. But its only incremented on CreateUpload, not for each part. So this is wrong. 

I think I'll fix that in a separate bug fix PR

@matthewfala

Reply:
Oh oops. I read this wrong. We do it only if we are removing the upload, so this is correct.